### PR TITLE
Breeder and Clinical Remarks ETL and reporting

### PR DESCRIPTION
**Pull request by labkey-martyp**
#### Rationale
Incremental ETLs migrating Breeder and Comments (Clinical Remarks) into datasets.  Update reporting including clinical history for clinical remarks.

#### Related Pull Requests
https://github.com/LabKey/ehrModules/pull/250

#### Changes
* ETL definitions and source queries
* Dataset definitions
* Views and metadata
* Reporting pages
* Clinical history for Clinical Remarks

### CODE REVIEW

File: nirc_ehr/resources/queries/dbo/q_breeder.sql [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+         LEFT JOIN ANIMAL_EVENT_COMMENT anmCmt ON anmEvt.ANIMAL_EVENT_ID = anmCmt.ANIMAL_EVENT_ID
+         LEFT JOIN EVENT_EVENT_GROUP evtEvtGrp ON evtEvtGrp.EVENT_ID = anmEvt.EVENT_ID
+         LEFT JOIN AUDIT_TRAIL adt
+                   ON anmEvt.ANIMAL_EVENT_ID = substring(adt.PRIMARY_KEY_VALUES, length('ANIMAL_EVENT_ID = '))
+                       AND adt.PRIMARY_KEY_VALUES LIKE '%ANIMAL_EVENT_ID%'
+
+WHERE evtEvtGrp.EVENT_GROUP_ID = 56
+--     56 Breeder
+
+GROUP BY anmEvt.ANIMAL_EVENT_ID,
```
Comment (ankurjuneja) [inline nirc_ehr/resources/queries/dbo/q_breeder.sql]: Don't see any aggregate column, Is this group by needed?

File: nirc_ehr/resources/schemas/dbscripts/postgresql/nirc_ehr-21.016-21.017.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+SELECT core.executeJavaUpgradeCode('reloadStudy');
+SELECT core.executeJavaUpgradeCode('etl;{NIRC_EHR}/breeder');
+SELECT core.executeJavaUpgradeCode('etl;{NIRC_EHR}/ehr_lookups');
```
Comment (ankurjuneja) [inline nirc_ehr/resources/schemas/dbscripts/postgresql/nirc_ehr-21.016-21.017.sql]: I think we might need to truncate ehr_lookups and run.

**Review by ankurjuneja — APPROVED** [review]

File: nirc_ehr/resources/schemas/dbscripts/postgresql/nirc_ehr-21.016-21.017.sql (hunk 2 above)
Comment (ankurjuneja) [inline nirc_ehr/resources/schemas/dbscripts/postgresql/nirc_ehr-21.016-21.017.sql]:
```suggestion
SELECT core.executeJavaUpgradeCode('etl;{NIRC_EHR}/ehr_lookups;truncate');
```

**Review by labkey-jeckels — APPROVED** [review]
Looks OK to me with Ankur's suggestions. Per our discussion, would be great to get the lookup sets cache to auto-clear via a DataIterator and/or QueryUpdateService hook

File: nirc_ehr/resources/schemas/dbscripts/postgresql/nirc_ehr-21.016-21.017.sql (hunk 2 above)
Comment (labkey-martyp) [reply nirc_ehr/resources/schemas/dbscripts/postgresql/nirc_ehr-21.016-21.017.sql]: Good call. Updated

File: nirc_ehr/resources/queries/dbo/q_breeder.sql (hunk 1 above)
Comment (labkey-martyp) [reply nirc_ehr/resources/queries/dbo/q_breeder.sql]: No, removed from both source queries. There are a few other source queries with group bys but no aggregates, wasn't sure if there were duplicate row issues in the source data.

**Comment by labkey-martyp** [comment]
> Looks OK to me with Ankur's suggestions. Per our discussion, would be great to get the lookup sets cache to auto-clear via a DataIterator and/or QueryUpdateService hook

@labkey-jeckels Added cache clearing in this PR, https://github.com/LabKey/ehrModules/pull/250

**Pull request merged**